app/workers/asset_manager_create_whitehall_asset_worker.rb

```diff
@@ -18,7 +18,9 @@ def perform(file_path, legacy_url_path, draft = false, attachable_model_class =
       end
     end
 
-    asset_manager.create_whitehall_asset(asset_options)
+    response = asset_manager.create_whitehall_asset(asset_options)
+
+    create_asset_manager_asset(model_id, asset_variant, response)
```

Comment:
I'm slightly nervous about this.

The code comment just a few lines below this says:

> The AttachmentData we want to set the timestamp on may not exist yet

which is why the job `AssetManagerAttachmentSetUploadedToWorker` runs asynchronously ½ second later – i.e. to make sure the `AttachmentData` model has been persisted to the database.

And if it hasn't been persisted yet, presumably it won't have an ID yet. So I'm sort of expecting `model_id` to be `nil`. In which case, I don't think we should be able to create an `Asset` at this point in time/code?

I saw your earlier commit which implemented this in the `AssetManagerAttachmentSetUploadedToWorker` – which makes more sense to me.

Am I missing something? 🤔

Reply:
One more thought: I wonder if this situation isn't occurring in test because Sidekiq jobs don't execute asynchronously. For example, the integration tests make use of `Sidekiq::Testing.inline!` to cause worker jobs to be executed synchronously.

Could it be possible that the `AttachmentData` record will always exist in tests, but not in the real world? 🤔